### PR TITLE
Added basic peer manager

### PR DESCRIPTION
**Pull request by jm-clius**
This is an incremental PR towards #358

It introduces a basic peer manager for `wakunode2`. For now the manager:
- abstracts the dialing of `WakuRelay` peers
- keeps track of dialed peers in a peer store
- provides an API to retrieve combined info from both the `peerStore` and `switch` about peers and their connectedness

### Suggested next steps:

1. general improvements 
2. use peer manager to present information about `relay` peers on the JSON-RPC Admin API
3. keep track locally of waku-related metadata for peers (incl. their connection status and behaviour). For this waku must implement its own extended peer store to add homegrown metadata types. Currently the manager just uses the "standard" peer store and assumes all peers must remain connected, as in a peer pool.
4. extend manager to be used for other Waku v2 protocols: `filter`, `store` and `swap` (requires _**3.**_)

### Future work:

1. Strategy to attempt redialing disconnected peers
2. Peer scoring mechanism
3. Peer store persistence
4. Peer discovery

### CODE REVIEW

**Review by staheri14 — APPROVED** [review]
Nice and clear comments on the code!

File: waku/v2/node/peer_manager.nim [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# Dialer interface #
+####################
+
+proc dialPeer*(pm: PeerManager, peerInfo: PeerInfo, proto: string): Future[Connection] {.async.} =
+  # Dial a given peer and add it to the list of known peers
+  # @TODO check peer validity, duplicates and score before continuing. Limit number of peers to be managed.
+  
+  # First add dialed peer info to peer store...
+
+  debug "Adding dialed peer to manager", peerId = peerInfo.peerId, addr = peerInfo.addrs[0], proto = proto
```
Comment (oskarth) [inline waku/v2/node/peer_manager.nim:27]: Might want to set a topic as log scope

File: waku/v2/node/peer_manager.nim [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+proc new*(T: type PeerManager, switch: Switch): PeerManager =
+  T(switch: switch,
+    peerStore: PeerStore.new())
+
+####################
+# Dialer interface #
+####################
+
+proc dialPeer*(pm: PeerManager, peerInfo: PeerInfo, proto: string): Future[Connection] {.async.} =
```
Comment (oskarth) [inline waku/v2/node/peer_manager.nim:21]: What happens if this call fails?

**Review by oskarth — APPROVED** [review]
In general looks good, some comments and suggestions

File: waku/v2/node/peer_manager.nim (hunk 1 above)
Comment (jm-clius) [reply waku/v2/node/peer_manager.nim:27]: Will do in subsequent PR.

File: waku/v2/node/peer_manager.nim (hunk 2 above)
Comment (jm-clius) [reply waku/v2/node/peer_manager.nim:21]: Currently still a very ungraceful unhandled `DialFailedError`. The plan is to timeout the `dial` future on the `switch`, log these failed/timed-out connections and keep track of such peers (e.g. changing their `connectedness` state to `CannotConnect` for a period of time). I'm also considering adding at least one redial attempt per failure, if there is a sensible way of doing this.

**Pull request merged**